### PR TITLE
Updated the `postgres` package to version `3.0.0`

### PR DESCRIPTION
**Pull request by BreX900**
- You no longer need to use the `Database` class, you can use `Connection.open` or `Pool` directly.
 - `Database` class implements postegres `Session` and `SessionExecutor` classes
 - Extension methods for accessing repositories now extend `Session`
 - Added the ability to create a Pool for your database

 **BREAKING CHANGES**
 - Updated the `postgres` package to version `3.0.0`. To visit all the breaking changes see their changelog
 - Removed `Database.query` method in favour of `Session.execute` method
 - `Action` and `Query` now accept a Session
 - Repository methods are no longer wrapped in a transaction
 - Removed all fields and methods to execute a transaction in favor of `SessionExecutor.runTx`

Blocked by #70 

Fixs #69 #5

### CODE REVIEW

**Comment by BreX900** [comment]
@schultek  Is there any possibility to merge the PR or is the repo no longer maintained?

**Comment by schultek** [comment]
Sry I haven't been showing much love to the package. I try to review the PR over the next days and get back to working on it some more.

File: lib/src/internals/base_repository.dart [hunk 1]
```diff
@@ -37,12 +37,8 @@ abstract class BaseRepository implements ModelRepository {
     return query.apply(db, params);
   }
 
-  Future<T> transaction<T>(Runnable<T> runnable) {
-    return db.runTransaction(runnable);
-  }
-
   @override
   Future<void> run<T>(Action<T> action, T request) {
-    return transaction(() => action.apply(db, request));
```
Comment (schultek) [inline lib/src/internals/base_repository.dart:46]: What is the reason for removing all the transactions? Data modifications should always be atomic and the package may run multiple queries for a insert or delete action.

Comment (BreX900) [reply lib/src/internals/base_repository.dart:46]: For various reasons:
- It should be the user who wants this behavior.
- The user so can perform multiple queries in a single transaction
- The Session class does not have the methods to execute transactions

Thanks to this change you can now write:
```dart
  final db = Database(...);
  await db.run((session) async {
    await session.companies.deleteOne('abc');

    await session.companies.insertOne(CompanyInsertRequest(
      id: 'abc',
      name: 'Minga',
      addresses: [],
    ));

    await session.accounts.deleteMany([0, 1, 2]);
  });
```
Remember: when you open one transaction you cannot open another so it must be the user who wants to open a transaction.

Comment (schultek) [reply lib/src/internals/base_repository.dart:46]: Ok thanks for the explanation

**Review by schultek — APPROVED** [review]
Thanks so much for your effort

**Pull request merged**